keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/PersistentEvent.kt

```diff
@@ -1,8 +1,20 @@
 package com.netflix.spinnaker.keel.events
 
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
 import java.time.Clock
 import java.time.Instant
 
+@JsonTypeInfo(
+  use = JsonTypeInfo.Id.NAME,
+  property = "type",
+  include = JsonTypeInfo.As.PROPERTY
+)
```

Comment:
isn't the discriminator actually the `scope` property? (in which case it would also be `include = As.EXISTING_PROPERTY`)

Reply:
It is, but it turns out that is not what `property` is used for by Jackson. It's just used to name the JSON property that will include the sub-type `name` as defined below. I was confused by this too until I read the docs again.

Reply:
Having said that, I did not read up on, or try, `As.EXISTING_PROPERTY`. Let me give it a try.

Reply:
I tried `AS.EXISTING_PROPERTY` with `scope` but that didn't work. For some reason it only works if I use `"type"` as the name of the property, which is the same name used in the `@JsonTypeInfo` annotation in the 2nd-level base classes. 🤷‍♂  

The reason I even added this here was so that Jackson could rebuild the class hierarchy (e.g. `PersistentEvent` -> `ResourceEvent` -> `ResourceCreated`), even though those base classes (which are, in fact, abstract) are never (de-)serialized themselves.

Reply:
Aren't we going to end up with both `type` and `scope` in the JSON holding the same value?

Reply:
No, because `type` is "overwritten" by the concrete sub-classes (which are the only ones that can actually be created and hence serialized). The output looks like the sample in the PR description.

Reply:
boy I was confused there. Looking at the sample it's clear they're different things